--- gamess/comp	2013-05-03 13:18:53.000000000 -0700
+++ patch-files/comp	2014-01-14 16:21:05.000000000 -0800
@@ -1,4 +1,4 @@
-#!/bin/csh
+#!/bin/csh -f
 #
 #  1 Feb 2013 - script to compile one GAMESS source file
 #  Invoke this script by './comp MODULE >& MODULE.log &'.
@@ -1833,7 +1833,7 @@
 #     -WB means warn but don't fail on out-of-bounds array references
 #     -ftz flushes underflow results to zero
 #
-      set OPT = '-O2'
+      set OPT = '-O2 -xHost'
       if (".$GMS_DEBUG_FLAGS" != .) set OPT="$GMS_DEBUG_FLAGS"
 #       alas, I'm so tired of hearing about so many versions failing to
 #       diagonalize, that this time critical code is run w/o optimization.
