--- gamess.2014.12/ddi/compddi	2015-01-15 06:49:05.000000000 -0800
+++ patch-files/compddi	2015-09-24 21:58:45.237940690 -0700
@@ -1,4 +1,4 @@
-#!/bin/csh
+#!/bin/csh -f
 #
 # 27 Mar 2014 - script to compile the Distributed Data Interface (DDI)
 #
@@ -15,7 +15,7 @@
    exit 4
 endif
 #
-set EXTRA_FLAGS=''
+#set EXTRA_FLAGS=''
 #
 #  the configuration file read just above provides the next two variables,
 #  and several other GMS_XXX values that will be used further down.
@@ -61,7 +61,7 @@
 #  along with collecting information about which flavor of mpi,
 #  where the MPI library is located, and so forth.
 #
-set COMM = $GMS_DDI_COMM
+set COMM = mixed
 #
 #     high end machines are just hardwired as follows.  Don't change them.
 #
@@ -616,7 +616,7 @@
 
      switch ($FORTRAN)
         case g77:
-        case pgfortran:
+#       case pgfortran:
         case pgf77:
         case f2c:
            set NumUS=2
@@ -675,7 +675,7 @@
            set NumUS=1
            set FORT_FLAGS = "-O2 $ARCH -fdefault-integer-8 -std=legacy"
            breaksw
-        case pgfortran
+#       case pgfortran
         case pgf77:
            set NumUS=1
            set CFLAGS = "$CFLAGS -mcmodel=medium"
@@ -906,7 +906,7 @@
 
      set FORTRAN=$GMS_FORTRAN     # choose from pgf77 and pgfortran
 
-        case pgfortran:
+#       case pgfortran:
         case pgf77:
            set CC = 'pgcc'
 #
@@ -958,7 +958,7 @@
            set NumUS=0
            set F77_OPTS = "-DF77_UPPERCASE -Dgetarg_=GETARG -Diargc_=IARGC"
            breaksw
-        case pgfortran:
+#       case pgfortran:
         case pgf77:
            set CC = 'pgcc'
 #
@@ -1064,7 +1064,8 @@
               )
     echo "Compiling common object: $OBJ.o"
     set echo
-    $CC $CFLAGS $EXTRA_FLAGS:q $DDI_OPTS -o $OBJ.o -c $OBJ.c
+#   $CC $CFLAGS $EXTRA_FLAGS:q $DDI_OPTS -o $OBJ.o -c $OBJ.c
+    $CC $CFLAGS $DDI_OPTS -o $OBJ.o -c $OBJ.c
     unset echo
     if(-e $OBJ.obj) mv -v $OBJ.obj $OBJ.o
     if(-e $OBJ.o) then
@@ -1124,7 +1125,8 @@
       $CC $WFLAGS $EXTRA_FLAGS:q $DDI_OPTS -o $OBJ.o -c $OBJ.c
     else
       set echo
-      $CC $CFLAGS $EXTRA_FLAGS:q $DDI_OPTS -o $OBJ.o -c $OBJ.c
+#     $CC $CFLAGS $EXTRA_FLAGS:q $DDI_OPTS -o $OBJ.o -c $OBJ.c
+      $CC $CFLAGS $DDI_OPTS -o $OBJ.o -c $OBJ.c
     endif
     unset echo
     if(-e $OBJ.obj) mv -v $OBJ.obj $OBJ.o
@@ -1146,7 +1148,8 @@
   chdir src
   echo "Compiling FORTRAN wrappers for DDI"
   set echo
-  $CC $CFLAGS $EXTRA_FLAGS:q $DDI_OPTS $F77_OPTS -o ddi_fortran.o -c ddi_fortran.c
+# $CC $CFLAGS $EXTRA_FLAGS:q $DDI_OPTS $F77_OPTS -o ddi_fortran.o -c ddi_fortran.c
+  $CC $CFLAGS $DDI_OPTS $F77_OPTS -o ddi_fortran.o -c ddi_fortran.c
   unset echo
   if(-e ddi_fortran.obj) mv -v ddi_fortran.obj ddi_fortran.o
   if(-e ddi_fortran.o) then
