--- gamess.2014.12/comp	2015-01-15 06:49:05.000000000 -0800
+++ patch-files/comp	2015-02-20 19:31:52.000000000 -0800
@@ -1740,7 +1740,7 @@
 #
 #         first, initialize OPT and EXTRAOPT
 #
-      set OPT='-O2'
+      set OPT='-O0'
       if ($MODULE == zheev)  set OPT='-O0'  # defensive compiling
 
       set EXTRAOPT=" "
@@ -1829,7 +1829,7 @@
 #     -WB means warn but don't fail on out-of-bounds array references
 #     -ftz flushes underflow results to zero
 #
-      set OPT = '-O2'
+      set OPT = '-O2 -xHost'
       if (".$GMS_DEBUG_FLAGS" != .) set OPT="$GMS_DEBUG_FLAGS"
 #       alas, I'm so tired of hearing about so many versions failing to
 #       diagonalize, that this time critical code is run w/o optimization.
