--- gamess-2018-09/comp	2018-09-30 20:45:26.000000000 -0700
+++ patch-files/comp	2019-02-11 14:39:07.779209744 -0800
@@ -1,4 +1,4 @@
-#!/bin/csh
+#!/bin/csh -f
 #
 #   5 Dec 2014 - script to compile one GAMESS source file
 #  Invoke this script by './comp MODULE >& MODULE.log &'.
@@ -1944,7 +1944,7 @@
 #
 #         first, initialize OPT and EXTRAOPT
 #
-      set OPT='-O2'
+      set OPT='-O0'
       if ($MODULE == zheev)  set OPT='-O0'  # defensive compiling
 
       set EXTRAOPT=" "
@@ -2109,7 +2109,7 @@
 #     -WB means warn but don't fail on out-of-bounds array references
 #     -ftz flushes underflow results to zero
 #
-      set OPT = '-O2'
+      set OPT = '-O2 -xHost'
       if (".$GMS_DEBUG_FLAGS" != .) set OPT="$GMS_DEBUG_FLAGS"
 #       alas, I'm so tired of hearing about so many versions failing to
 #       diagonalize, that this time critical code is run w/o optimization.
