--- gamess/comp	2013-05-03 13:18:53.000000000 -0700
+++ patch-files/comp	2014-09-15 11:20:41.668204515 -0700
@@ -1,4 +1,4 @@
-#!/bin/csh
+#!/bin/csh -f
 #
 #  1 Feb 2013 - script to compile one GAMESS source file
 #  Invoke this script by './comp MODULE >& MODULE.log &'.
@@ -1755,7 +1755,7 @@
 #           -finit-real=<zero|inf|-inf|nan>  could be interesting debug opt
 #           -fno-automatic forces static storage of all local variables
 #           -fno-whole-file suppresses argument's data type checking
-      set OPT='-O2'
+      set OPT='-O0'
       if (".$GMS_DEBUG_FLAGS" != .) set OPT="$GMS_DEBUG_FLAGS"
 
       if ($GMS_GFORTRAN_VERNO == 4.6) then
@@ -1833,7 +1833,7 @@
 #     -WB means warn but don't fail on out-of-bounds array references
 #     -ftz flushes underflow results to zero
 #
-      set OPT = '-O2'
+      set OPT = '-O2 -xHost'
       if (".$GMS_DEBUG_FLAGS" != .) set OPT="$GMS_DEBUG_FLAGS"
 #       alas, I'm so tired of hearing about so many versions failing to
 #       diagonalize, that this time critical code is run w/o optimization.
