--- gamess-2018-09/ddi/compddi	2018-08-07 00:23:20.000000000 -0700
+++ patch-files/compddi	2019-02-11 14:45:54.669255295 -0800
@@ -1,4 +1,4 @@
-#!/bin/csh
+#!/bin/csh -f
 #
 # 27 Mar 2014 - script to compile the Distributed Data Interface (DDI)
 #
@@ -61,7 +61,7 @@
 #  along with collecting information about which flavor of mpi,
 #  where the MPI library is located, and so forth.
 #
-set COMM = $GMS_DDI_COMM
+set COMM = mixed
 #
 #     high end machines are just hardwired as follows.  Don't change them.
 #
@@ -702,7 +702,7 @@
 
      switch ($FORTRAN)
         case g77:
-        case pgfortran:
+#       case pgfortran:
         case pgf77:
         case f2c:
            set NumUS=2
@@ -769,7 +769,7 @@
            set NumUS=1
            set FORT_FLAGS = "-O2 $ARCH -fdefault-integer-8 -std=legacy"
            breaksw
-        case pgfortran
+#       case pgfortran
         case pgf77:
            set NumUS=1
            if ( `uname` == Darwin ) then
@@ -1005,7 +1005,7 @@
 
      set FORTRAN=$GMS_FORTRAN     # choose from pgf77 and pgfortran
 
-        case pgfortran:
+#       case pgfortran:
         case pgf77:
            set CC = 'pgcc'
 #
@@ -1057,7 +1057,7 @@
            set NumUS=0
            set F77_OPTS = "-DF77_UPPERCASE -Dgetarg_=GETARG -Diargc_=IARGC"
            breaksw
-        case pgfortran:
+#       case pgfortran:
         case pgf77:
            set CC = 'pgcc'
 #
@@ -1164,7 +1164,7 @@
               )
     echo "Compiling common object: $OBJ.o"
     set echo
-    $CC $CFLAGS $EXTRA_FLAGS:q $DDI_OPTS -o $OBJ.o -c $OBJ.c
+    $CC $CFLAGS $DDI_OPTS -o $OBJ.o -c $OBJ.c
     unset echo
     if(-e $OBJ.obj) mv -v $OBJ.obj $OBJ.o
     if(-e $OBJ.o) then
@@ -1221,10 +1221,10 @@
       set WFLAGS = '-DLINUX -DWINDOWS -DWINDOWS64 -DWINTEL -DOLDDDITIMER'
       set WFLAGS = "$WFLAGS -Od $GMS_WIN_TP -I./include"
       set echo
-      $CC $WFLAGS $EXTRA_FLAGS:q $DDI_OPTS -o $OBJ.o -c $OBJ.c
+      $CC $WFLAGS $DDI_OPTS -o $OBJ.o -c $OBJ.c
     else
       set echo
-      $CC $CFLAGS $EXTRA_FLAGS:q $DDI_OPTS -o $OBJ.o -c $OBJ.c
+      $CC $CFLAGS $DDI_OPTS -o $OBJ.o -c $OBJ.c
     endif
     unset echo
     if(-e $OBJ.obj) mv -v $OBJ.obj $OBJ.o
@@ -1246,7 +1246,7 @@
   chdir src
   echo "Compiling FORTRAN wrappers for DDI"
   set echo
-  $CC $CFLAGS $EXTRA_FLAGS:q $DDI_OPTS $F77_OPTS -o ddi_fortran.o -c ddi_fortran.c
+  $CC $CFLAGS $DDI_OPTS $F77_OPTS -o ddi_fortran.o -c ddi_fortran.c
   unset echo
   if(-e ddi_fortran.obj) mv -v ddi_fortran.obj ddi_fortran.o
   if(-e ddi_fortran.o) then
