--- gamess/ddi/src/ddi_create.c	2013-05-03 13:14:15.000000000 -0700
+++ patch-files/ddi_create.c	2014-01-27 18:13:45.000000000 -0800
@@ -1,7 +1,7 @@
 /* -------------------------------------------------------------------- *\
  * Distributed Data Interface
  * ==========================
- * 
+ *
  * Collective operations used to create a distributed array.
  *
  * Author: Ryan M. Olson
@@ -9,8 +9,8 @@
  * 29 Mar 10 - SS  - add missing 3rd argument to ddi-send-request
  * 18 Aug 10 - MWS - use 64 bit integer for printing of DM total size
  * 18 Feb 13 - MWS - trap for >2 Gword MEMDDI slice size
-\* -------------------------------------------------------------------- */
- # include "ddi_base.h"
+ * 31 Jul 13 - MWS - fix slice size check for TCP/IP runs
+\* -------------------------------------------------------------------- */  # include "ddi_base.h"
 
 /* -------------------------------------------------------------------- *\
    DDI_Create(idim,jdim,handle)
@@ -132,10 +132,15 @@
       totwrds   = longrows*longcols;
    /*     Total distributed array over 2 Gwords is OK, but each  */
    /*     slice (MEMDDI per data server) must be under 2 GWords. */
-   /*     TCP/IP on one node has gv(nd)=-1 since no d.s. exists, */
+   /*     TCP/IP has gv(nd)=-1 (uninitialized)                   */
    /*     Cray on one node has gv(nd)=0 since no d.s. exists.    */
-      longnd    = gv(nd);
-      if (longnd <= 0) longnd=1;
+      # if defined DDI_MPI
+         longnd    = gv(nd);
+         if (longnd <= 0) longnd=1;
+      # endif
+      # if defined DDI_SOC
+         longnd = np;
+      # endif
       longslice = totwrds/longnd;
    /*  next is largest signed 32 bit integer, stored as 64 bit quantity  */
       long2g   = 2147483643;
