--- gamess-2019.06/gms-files.csh	2019-08-04 10:27:44.000000000 -0700
+++ patch-files/gms-files.csh	2019-09-06 10:18:14.711941659 -0700
@@ -48,7 +48,7 @@
 setenv OPTHES1 $USERSCR/$JOB.hs1
 setenv OPTHES2 $USERSCR/$JOB.hs2
 setenv     XYZ $SCR/$JOB.xyz
-setenv   INPUT $SCR/$JOB.F05
+setenv   NEWINPUT $SCR/$JOB.F05
 setenv   PUNCH $USERSCR/$JOB.dat
 setenv  AOINTS $SCR/$JOB.F08
 setenv  MOINTS $SCR/$JOB.F09
