--- gamess-2018-09/gms-files.csh	2018-09-30 20:45:26.000000000 -0700
+++ patch-files/gms-files.csh	2019-02-11 14:50:49.501279421 -0800
@@ -48,7 +48,7 @@
 setenv OPTHES1 $USERSCR/$JOB.hs1
 setenv OPTHES2 $USERSCR/$JOB.hs2
 setenv     XYZ $SCR/$JOB.xyz
-setenv   INPUT $SCR/$JOB.F05
+setenv   NEWINPUT $SCR/$JOB.F05
 setenv   PUNCH $USERSCR/$JOB.dat
 setenv  AOINTS $SCR/$JOB.F08
 setenv  MOINTS $SCR/$JOB.F09
