--- gamess/lked	2013-05-17 09:15:06.000000000 -0700
+++ patch-files/lked	2014-09-12 10:07:35.250410263 -0700
@@ -1,4 +1,4 @@
-#!/bin/csh
+#!/bin/csh -f
 #
 # 17 May 2013 - script to "link-edit" a GAMESS executable
 #
@@ -435,6 +435,7 @@
           breaksw
         case 11:
         case 12:
+        case 13:
           # next attempts a static link, whereas 10 above is a dynamic link.
           # if this fails to work, try the version 10 lines shown just above.
           set MATHLIBS="-Wl,--start-group"
@@ -548,6 +549,7 @@
         case 10:
         case 11:
         case 12:
+        case 13:
           # next are a static link, in single-threaded mode (serial BLAS).
           # The choices preclude any need to
           #     a) install .so shared object libraries on every node,
@@ -972,6 +974,7 @@
       set MPILIBS=''
       if ($TARGET == linux64) then
         if (($GMS_DDI_COMM == mpi) || ($GMS_DDI_COMM == mixed)) then
+          setenv LD_LIBRARY_PATH $GMS_MPI_PATH/lib:$LD_LIBRARY_PATH
           switch ($GMS_MPI_LIB)
             case impi:
                set MPILIBS="-L$GMS_MPI_PATH/lib64"
@@ -1009,12 +1012,14 @@
 #                   a new cluster needed to also look into -libnetdisc.
 #                   Finally, allow it to peek into the runtime/thread libs 
 #
-               set OFEDLIST="-libverbs -lrdmacm -libumad"
-               if (-e $GMS_OFED_PATH/libibnetdisc.so ) then
-                  set OFEDLIST="$OFEDLIST -libnetdisc"
-               endif
-               set MPILIBS="$MPILIBS -L$GMS_OFED_PATH $OFEDLIST"
-               set MPILIBS="$MPILIBS -lrt -lpthread"
+#               set OFEDLIST="-libverbs -lrdmacm -libumad"
+                set OFEDLIST="-libverbs -libmad -lrdmacm -libumad"
+#              if (-e $GMS_OFED_PATH/libibnetdisc.so ) then
+#                 set OFEDLIST="$OFEDLIST -libnetdisc"
+#              endif
+#              set MPILIBS="$MPILIBS -L$GMS_OFED_PATH $OFEDLIST"
+               set MPILIBS="$MPILIBS -L$GMS_MPI_PATH/lib $OFEDLIST"
+               set MPILIBS="$MPILIBS -llimic2 -lrt -lpthread"
                breaksw
 
             case myrinet:
@@ -1023,7 +1028,7 @@
                set MPILIBS="$MPILIBS /usr/local/gm/lib/libgm.a"
                breaksw
             case openmpi:
-               set MPILIBS="-L$GMS_MPI_PATH/lib64"
+               set MPILIBS="-L$GMS_MPI_PATH/lib"
                set MPILIBS="$MPILIBS -lmpi"
                breaksw
             case mpt:
