--- gamess-2018-09/lked	2018-09-30 20:45:26.000000000 -0700
+++ patch-files/lked	2019-02-11 15:04:10.270326406 -0800
@@ -409,6 +409,7 @@
           breaksw
         case 11:
         case 12:
+        case 2018*:
           # next attempts a static link, whereas 10 above is a dynamic link.
           # if this fails to work, try the version 10 lines shown just above.
           set MATHLIBS="-Wl,--start-group"
@@ -575,6 +576,7 @@
         case 10:
         case 11:
         case 12:
+        case 2018*:
           # next are a static link, in single-threaded mode (serial BLAS).
           # The choices preclude any need to
           #     a) install .so shared object libraries on every node,
@@ -599,6 +601,7 @@
         case 10-so:
         case 11-so:
         case 12-so:
+        case 2018*-so:
           #  next one chooses dynamic linking (so= shared object libs)
           #  Just hand edit GMS_MKL_VERNO in your 'install.info' to use this.
           #  run-time libs (and LD_LIBRARY_PATH) needed on every compute node!
@@ -1048,6 +1051,7 @@
       set MPILIBS=''
       if ($TARGET == linux64) then
         if (($GMS_DDI_COMM == mpi) || ($GMS_DDI_COMM == mixed)) then
+          setenv LD_LIBRARY_PATH $GMS_MPI_PATH/lib:$LD_LIBRARY_PATH
           switch ($GMS_MPI_LIB)
             case impi:
                set MPILIBS="-L$GMS_MPI_PATH/lib64"
@@ -1079,42 +1083,9 @@
                   case 1.7:
                   case 1.8:
                   case 1.9:
-#                          several of our older clusters allow a static link
-                     set MPILIBS="$GMS_MPI_PATH/lib/libmpich.a"
-                     set MPILIBS="$MPILIBS $GMS_MPI_PATH/lib/libopa.a"
-                     set MPILIBS="$MPILIBS $GMS_MPI_PATH/lib/libmpl.a"
-#                          but you can try a .so link if you like:
-                     #--set MPILIBS="-L$GMS_MPI_PATH/lib -lmpich -lopa -lmpl"
-#
-#                   QLogic Infiniband likes to look at its
-#                   its "path scale messaging" PSM library.
-#                   Mellanux Infiniband does not need this library,
-#                   and it can be an error to name if non-existent.
-#
-                    if (-e /usr/lib64/libpsm_infinipath.so) then
-                       set MPILIBS="$MPILIBS -L/usr/lib64 -lpsm_infinipath"
-                    endif
-#
-#                   Append OFED's libs to the end of the MVAPICH2 list,
-#                   one particular cluster needed to look into -libnetdisc.
-#                   Finally, allow it to peek into the runtime/thread libs
-#
-                    set OFEDLIST="-libverbs -lrdmacm -libumad"
-                    if (-e $GMS_OFED_PATH/libibnetdisc.so ) then
-                       set OFEDLIST="$OFEDLIST -libnetdisc"
-                    endif
-                    set MPILIBS="$MPILIBS -L$GMS_OFED_PATH $OFEDLIST"
-                    set MPILIBS="$MPILIBS -lrt -lpthread"
-                    breaksw
-#
-#                       as of version 2.0, MVAPICH2 seems to have a simple
-#                       link list, just its own two libs.  It then finds
-#                       the requisite OFED libraries all on its own.
-#                       On our 2.1 cluster, DDI linked to various libs:
-#                           xml2, ibmad, rdmacm, ibumad, ibverbs, rt
-#                       without any need to specify them here.
                   case 2.0:
                   case 2.1:
+                     set MPILIBS="$MPILIBS ROLLMPILIBS"
                   default:
                      set MPILIBS="-L$GMS_MPI_PATH/lib -lmpifort -lmpi"
                      set MPILIBS="$MPILIBS -lpthread"
@@ -1122,8 +1093,7 @@
                   endsw
                breaksw
             case openmpi:
-               set MPILIBS="-L$GMS_MPI_PATH/lib"
-               set MPILIBS="$MPILIBS -lmpi"
+               set MPILIBS="$MPILIBS ROLLMPILIBS"
                breaksw
             case mpt:
                set MPILIBS="-L$GMS_MPI_PATH/lib"
