--- gamess.2014.12/rungms	2014-12-10 08:01:06.000000000 -0800
+++ patch-files/rungms	2015-05-19 10:27:08.617035851 -0700
@@ -84,6 +84,7 @@
 #      The SCHED variable, and scheduler assigned work space, is used
 #      below only in the MPI section.  See that part for more info.
                       set SCHED=none
+if ($?SLURM_JOB_ID)   set SCHED=SLURM
 if ($?PBS_O_LOGNAME)  set SCHED=PBS
 if ($?SGE_O_LOGNAME)  set SCHED=SGE
 if ($SCHED == SGE) then
@@ -101,6 +102,10 @@
    echo "PBS has assigned the following compute nodes to this run:"
    uniq $PBS_NODEFILE
 endif
+if ($SCHED == SLURM) then
+   echo "SLURM has assigned the following compute nodes to this run:"
+   export PBS_NODEFILE=`generate_pbs_nodefile`
+   uniq $PBS_NODEFILE
 #
 echo "Available scratch disk space (Kbyte units) at beginning of the job is"
 df -k $SCR
@@ -415,7 +420,7 @@
 #          names into the HOSTLIST string for the kickoff program,
 #          and to request the host name of the fast network adapters.
 #
-    if ($?PBS_JOBID) then
+    if ($?PBS_JOBID || $?SLURM_JOB_ID) then
 #
 #         The IBM cluster has two Gigabit adapters in each 4-way SMP,
 #         while the AXP cluster is based on a Myrinet network.
@@ -660,7 +665,7 @@
          set NNODES=`wc -l $HOSTFILE`
          set NNODES=$NNODES[1]
       endif
-      if ($SCHED == PBS) then
+      if ($SCHED == PBS || $SCHED == SLURM) then
          uniq $PBS_NODEFILE $HOSTFILE
          set NNODES=`wc -l $HOSTFILE`
          set NNODES=$NNODES[1]
